### ISSUE TITLE
Deleting volume ERROR

### ISSUE THREAD

**Issue by mluscon** [terraform-providers/terraform-provider-digitalocean#27]
Hi,

Terraform Version
```
Terraform v0.9.11
```

with this simple recipe:
```
provider "digitalocean" {
  token = "${var.digitalocean_token}"
}

resource "digitalocean_volume" "docker_bootstrap" {
  region      = "${var.region}"
  name        = "docker"
  size        = 2
  description = "docker volume"
}

resource "digitalocean_droplet" "dcos_bootstrap" {
  name = "${format("${var.dcos_cluster_name}-bootstrap-%02d", count.index)}"

  image = 25718515
  size = "${var.boot_size}"
  depends_on = ["digitalocean_volume.docker_bootstrap"]
  volume_ids = ["${digitalocean_volume.docker_bootstrap.id}"]
  ssh_keys = ["${var.ssh_key_fingerprint}"]
  private_networking = true
  connection {
    user = "root"
    private_key = "${file(var.dcos_ssh_key_path)}"
  }
  user_data     = "#cloud-config\n\nssh_authorized_keys:\n  - \"${file("${var.dcos_ssh_public_key_path}")}\"\n"
  region      = "${var.region}"
}
```
After successful run of `terraform apply` and a consequent run of `terraforom destroy` I get the following error:
```
Error applying plan:

1 error(s) occurred:

* digitalocean_volume.docker_bootstrap (destroy): 1 error(s) occurred:

* digitalocean_volume.docker_bootstrap: Error deleting volume: DELETE https://api.digitalocean.com/v2/volumes/ed9d6674-7389-11e7-9bdc-0242ac110704: 409 A volume that's attached to a Droplet cannot be deleted. Please detach it first before deleting.

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
There is probably a timing issue because the order of destroying seems to be legit.
```
digitalocean_volume.docker_bootstrap: Refreshing state... (ID: ed9d6674-7389-11e7-9bdc-0242ac110704)
digitalocean_droplet.dcos_bootstrap: Refreshing state... (ID: 56571893)
digitalocean_droplet.dcos_bootstrap: Destroying... (ID: 56571893)
digitalocean_droplet.dcos_bootstrap: Still destroying... (ID: 56571893, 10s elapsed)
digitalocean_droplet.dcos_bootstrap: Destruction complete
digitalocean_volume.docker_bootstrap: Destroying... (ID: ed9d6674-7389-11e7-9bdc-0242ac110704)
```

**Comment by stack72** [comment]
@grubernaut I believe this is closed via #22 but requires a release

**Comment by grubernaut** [comment]
@stack72 Indeed it looks like it was. 
@mluscon, just released v0.1.2 of the DigitalOcean provider for TF v0.10.0+. 
Can fetch the latest provider binary here: https://releases.hashicorp.com/terraform-provider-digitalocean/0.1.2

Thanks!
